sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -140,11 +144,12 @@ public AmqpAnnotatedMessage getAmqpAnnotatedMessage() {
      *
      * @return A byte array representing the data.
      */
-    public byte[] getBody() {
+    public BinaryData getBody() {
         final AmqpBodyType type = amqpAnnotatedMessage.getBody().getBodyType();
         switch (type) {
             case DATA:
-                return Arrays.copyOf(binaryData, binaryData.length);
+                return BinaryData.fromBytes(((AmqpDataBody) amqpAnnotatedMessage.getBody())
```

Comment:
This returns an optional. We should actually check if there are none in findFirst().